src/jwt/signatureVerifier.js

```diff
@@ -44,21 +45,30 @@ export const verifySignature = (idToken, options) => {
     return Promise.resolve(payload);
   }
 
-  return getJwk(options.domain, header.kid).then(jwk => {
-    const pubKey = KEYUTIL.getKey(jwk);
-    const signatureValid = KJUR.jws.JWS.verify(idToken, pubKey, ['RS256']);
+  return getJwk(options.domain, header.kid)
+    .then(jwk => rsaVerifierForKey(jwk))
+    .then(rsaVerifier => {
+      const encodedParts = idToken.split('.');
+      const headerAndPayload = encodedParts[0] + '.' + encodedParts[1];
+      const signature = base64.decodeToHEX(encodedParts[2]);
```

Comment:
Is there some prior verification that the ID Token has 3 parts?

Reply:
Yes, it's done at the beginning of `verifySignature` when jwtDecoder is called